pkg/image/util/signature/signature.go

```diff
@@ -0,0 +1,49 @@
+package signature
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+
+	// This enables the containers/image docker transport
+	_ "github.com/containers/image/docker"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+
+	imageapi "github.com/openshift/origin/pkg/image/apis/image"
+)
+
+// defaultTransportName is the name of default containers/image transport to use
+// to pull signatures.
+const defaultTransportName = "docker"
+
+// GetSignatureForImage pulls down image signatures from remote or local signature store.
+// See this document for more details about how to configure the signature stores:
+// https://github.com/containers/image/blob/master/docs/registries.d.md
+func GetSignatureForImage(ctx context.Context, registryHost, repositoryName, imageName string) ([]imageapi.ImageSignature, error) {
+	t := transports.Get(defaultTransportName)
```

Comment:
If `defaultTransportName` is supposed to be configurable, `transports.Get` may return `nil`.

But the way the reference is being created, it assumes the `docker` transport; so, just use `containers/image/docker.ParseReference` directly.

Reply:
thanks will try that